generator/definition.go

```diff
@@ -18,29 +19,59 @@ func NewDefinitionGenerator(definitionMap map[string]definition.Definition) *Def
 		Definitions: definitionMap,
 	}
 }
+func (g DefinitionGenerator) getDefiniationName(t interface{}) string {
+	hash := utils.GenerateHash(utils.ConvertToString(t))
+	// create random names for maps
+	if reflect.TypeOf(t).Kind() == reflect.Map && reflect.TypeOf(t).Name() == "" {
+		return strings.ReplaceAll(fmt.Sprintf("%T_%s_%s", t, utils.GetHashOfMap(utils.ConvertInterfaceToMap(t)), hash), "[]", "")
+	}
+	return strings.ReplaceAll(fmt.Sprintf("%T_%s", t, hash), "[]", "")
+}
 
-func (g DefinitionGenerator) CreateDefinition(t interface{}) {
+func (g DefinitionGenerator) CreateDefinition(t interface{}, prefix string) {
```

Comment:
Yeah, refactor would be nice here once the time comes. I recommend we use the 'Chain Of Responsibility' pattern for this. It follows a lot of the SOLID principles and would make this look a lot more cleaner and easy to test.